crmsh/cibconfig.py

```diff
@@ -816,26 +816,59 @@ def postprocess_cli(node, oldnode=None, id_hint=None):
     resolve_references(node)
     if oldnode is not None:
         remove_id_used_attributes(oldnode)
+    if complete_advised:
+        complete_advised_meta(node)
     return node, obj_type, obj_id
 
 
+def complete_advised_meta(node):
+    """
+    Complete advised meta attributes
+    """
+    if node.tag != "clone":
+        return
+    primitive_list = node.xpath('primitive')
+    if not primitive_list:
+        return
+    set_list = []
+    for meta_item in ["promotable", "interleave"]:
+        if not is_attr_set(node, meta_item):
+            set_list.append(meta_item)
+    if not set_list:
+        return
+
+    p = primitive_list[0]
+    ra_inst = RAInfo(p.get('class'), p.get('type'), p.get('provider'))
+    ra_actions_dict = ra_inst.actions()
+    if ra_actions_dict and "promote" in ra_actions_dict and "demote" in ra_actions_dict:
+        meta_node = get_set_nodes(node, "meta_attributes", create=True)[0]
+        for item in set_list:
+            set_attr(meta_node, item, "true")
+    # Add interleave=true as long as it's not set, no matter if it's promotable clone or not
+    elif "interleave" in set_list:
+        meta_node = get_set_nodes(node, "meta_attributes", create=True)[0]
+        set_attr(meta_node, "interleave", "true")
```

Comment:
Somehow if with a consolidated logic, we shouldn't have to call set_attr() for "interleave" separately for different cases, but I understand this way also works :-)